core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -845,10 +856,25 @@ oppia.factory('explorationStatesService', [
       }
     };
 
+    var convertToBackend = function(propertyValue) {
+      var propToBackend = [];
+      for (var i = 0; i < propertyValue.length; i++) {
```

Comment:
The modelling here seems off. The function seems to assume that propertyValue is a list, so it's somewhat specific to answer groups whilst purporting to be a general method.

Perhaps instead you could replace PROPERTIES_TO_CONVERT by a dict called BACKEND_CONVERSIONS. The keys are the property names and the values are the conversion functions. Then, if a property name (like answer_groups) appears in that dict, apply the corresponding function to its value prior to saving in the backend.